### PR TITLE
Static extract fixes bl

### LINKED ISSUE #401: Make content configureable for FurtherInformation
In the moment the content for FurtherInformation is not in the Report.properties file. It is hardcoded in template:

https://github.com/camptocamp/pyramid_oereb/blob/master/print/print-apps/oereb/toc.jrxml#L310

### PR DESCRIPTION
**Pull request by vvmruder**
fixes #401

### CODE REVIEW

**Comment by vvmruder** [comment]
@voisardf Can you implement that content of this is read from Report.properties? I dont know how to achieve this!

File: print/print-apps/oereb/toc.jrxml [hunk 1]
```diff
@@ -301,13 +301,12 @@
...
-				<textElement verticalAlignment="Top" markup="none">
-					<font fontName="Cadastra" size="6" isBold="true"/>
+			<textField isBlankWhenNull="true">
+				<reportElement x="260" y="105" width="232" height="11" uuid="423aa9b7-82cf-47fe-8581-6ac646500ca7"/>
+				<textElement verticalAlignment="Top" markup="html">
+					<font fontName="Cadastra" size="6"/>
 				</textElement>
-				<textFieldExpression><![CDATA["www.ne.ch/sgrf"]]></textFieldExpression>
-				<hyperlinkReferenceExpression><![CDATA["www.ne.ch/sgrf"]]></hyperlinkReferenceExpression>
+				<textFieldExpression><![CDATA["Weitere Informationen zum ÖREB-Kataster finden Sie auf unserer <a href=\"https://www.baselland.ch/politik-und-behorden/direktionen/volkswirtschafts-und-gesundheitsdirektion/amt-fur-geoinformation/oereb-kataster\"><font color=\"blue\">Webseite</font></a>"]]></textFieldExpression>
```
Comment (sbrunner) [inline print/print-apps/oereb/toc.jrxml]: Shouldn't be put in a print attribute as default value, or in the application config...

Comment (vvmruder) [reply print/print-apps/oereb/toc.jrxml]: Yes. See my comment above.

Comment (vvmruder) [reply print/print-apps/oereb/toc.jrxml]: It was hard coded until now. I want to change it to be read from Report.properties cause it is a cantonal but static content.

Comment (vvmruder) [reply print/print-apps/oereb/toc.jrxml]: @sbrunner: Can you make help out with reading the content of this field out of Report.properties?

Comment (sbrunner) [reply print/print-apps/oereb/toc.jrxml]: See sample PR :-)

Comment (vvmruder) [reply print/print-apps/oereb/toc.jrxml]: ?

Comment (sbrunner) [reply print/print-apps/oereb/toc.jrxml]: https://github.com/camptocamp/pyramid_oereb/pull/403 :-)

Comment (vvmruder) [reply print/print-apps/oereb/toc.jrxml]: Ahhh :P

**Comment by vvmruder** [comment]
I added it to the properties of the report. It is a static string somehow. But different between cantons. @sbrunner: what do you think? Will it be ok this way? Or should I really go the way through config? Because, this is only affecting the PDF

**Comment by vvmruder** [comment]
@voisardf & @kdeininger : What is your opinion on this

**Comment by sbrunner** [comment]
The `Report.properties` is the localisation of the print...
If is should be changed the advantage of putting t in the config is that we shouldn't change the print app in the project instance. But for me it's acceptable like that.

**Comment by vvmruder** [comment]
Ok. I will adapt it to the yaml config way. You are right. :)

**Comment by vvmruder** [comment]
@sbrunner:
I got this error:
```
Error while processing request:
com.vividsolutions.jts.util.AssertionFailedException: The following parameters are declared in pdfextract.jrxml but are not output values of processors or attributes.
Either remove the references or update the configuration so that all the parameters are available for the report.

	* Further_Information_Text

```

File: print/print-apps/oereb/toc.jrxml [hunk 2]
```diff
@@ -306,7 +307,7 @@
 				<textElement verticalAlignment="Top" markup="html">
 					<font fontName="Cadastra" size="6"/>
 				</textElement>
-				<textFieldExpression><![CDATA[$R{FurtherInformationContent}]]></textFieldExpression>
+				<textFieldExpression><![CDATA[$P{Further_Information_Text}]]></textFieldExpression>
```
Comment (sbrunner) [inline print/print-apps/oereb/toc.jrxml]: I prefer camel case...
Actually we use the _ in the print attributes when we flatten the attributes of an object `<Object>_<Attribute>`.

File: pyramid_oereb/standard/pyramid_oereb.yml.mako [hunk 3]
```diff
@@ -84,6 +84,11 @@ pyramid_oereb:
     # The headers sent to the print
     headers:
       Content-Type: application/json; charset=UTF-8
+    furtherInfromationText:
+      de: [Hier sollten die weiteren Informationen erfasst werden.]
```
Comment (sbrunner) [inline pyramid_oereb/standard/pyramid_oereb.yml.mako]: Your sure about the [], It will mean that we get an array with the words...

Comment (vvmruder) [reply pyramid_oereb/standard/pyramid_oereb.yml.mako]: The "Further_Information_Text" attribute is in the JSON when I call the service with "getspec". This means to me, that there is wrong something else. Do I need to configure the new parameter somwhere else?
@voisardf: Maybe you can help out here?

Comment (sbrunner) [reply pyramid_oereb/standard/pyramid_oereb.yml.mako]: @vvmruder I don't understand what's wrong ...

Comment (vvmruder) [reply pyramid_oereb/standard/pyramid_oereb.yml.mako]: Öhhh... You saw the error I posted above?

I think this is caused by mapfish print? What is normally causing this error?

**Comment by vvmruder** [comment]
I think I solved it. I changed <![CDATA[$P{furtherInfromationText}]]> to <![CDATA[$R{furtherInfromationText}]]> in the toc.jrxml and now it works.

**Comment by sbrunner** [comment]
Than you get it from the localisation, not from the attributes ...

File: pyramid_oereb/standard/pyramid_oereb.yml.mako (hunk 3 above)
Comment (sbrunner) [reply pyramid_oereb/standard/pyramid_oereb.yml.mako]: No idea about the error, for me everything looks correct ...

Comment (voisardf) [reply pyramid_oereb/standard/pyramid_oereb.yml.mako]: what about the brackets? could that be the issue in the yaml?

File: print/print-apps/oereb/config.yaml [hunk 4]
```diff
@@ -249,6 +249,7 @@ templates:
                                 strokeOpacity: 1
                                 strokeWidth: 2
                                 strokeLinecap: "round"
+            furtherInfromationText: !string {}
```
Comment (voisardf) [inline print/print-apps/oereb/config.yaml]: spelling error:  infROmation instead of infORmation

File: print/print-apps/oereb/pdfextract.jrxml [hunk 5]
```diff
@@ -50,6 +50,7 @@
 	<parameter name="RealEstate_FosNr" class="java.lang.String"/>
 	<parameter name="RealEstate_RestrictionOnLandownershipDataSource" class="net.sf.jasperreports.engine.JRDataSource"/>
 	<parameter name="GlossaryDataSource" class="net.sf.jasperreports.engine.JRDataSource"/>
+    <parameter name="furtherInfromationText" class="java.lang.String"/>
```
Comment (voisardf) [inline print/print-apps/oereb/pdfextract.jrxml]: please correct spelling, cheers

File: print/print-apps/oereb/pdfextract.jrxml [hunk 6]
```diff
@@ -228,6 +229,9 @@
 					<subreportParameter name="REPORT_RESOURCE_BUNDLE">
 						<subreportParameterExpression><![CDATA[$P{REPORT_RESOURCE_BUNDLE}]]></subreportParameterExpression>
 					</subreportParameter>
+					<subreportParameter name="furtherInfromationText">
```
Comment (voisardf) [inline print/print-apps/oereb/pdfextract.jrxml]: same :)

File: print/print-apps/oereb/pdfextract.jrxml [hunk 7]
```diff
@@ -228,6 +229,9 @@
...
+						<subreportParameterExpression><![CDATA[$P{furtherInfromationText}]]></subreportParameterExpression>
```
Comment (voisardf) [inline print/print-apps/oereb/pdfextract.jrxml]: again ;)

File: print/print-apps/oereb/toc.jrxml [hunk 8]
```diff
@@ -75,6 +75,7 @@
 	<parameter name="RealEstate_FosNr" class="java.lang.String"/>
 	<parameter name="RealEstate_LandRegistryArea" class="java.lang.String"/>
 	<parameter name="RealEstate_SubunitOfLandRegister" class="java.lang.String"/>
+    <parameter name="furtherInfromationText" class="java.lang.String"/>
```
Comment (voisardf) [inline print/print-apps/oereb/toc.jrxml]: replace :)

**Comment by vvmruder** [comment]
Ohhh noooo...

Ok then I need to investigate more in detail :(

File: print/print-apps/oereb/toc.jrxml [hunk 9]
```diff
@@ -301,13 +302,12 @@
...
-				<textElement verticalAlignment="Top" markup="none">
-					<font fontName="Cadastra" size="6" isBold="true"/>
+			<textField isBlankWhenNull="true">
+				<reportElement x="260" y="105" width="232" height="11" uuid="423aa9b7-82cf-47fe-8581-6ac646500ca7"/>
+				<textElement verticalAlignment="Top" markup="html">
+					<font fontName="Cadastra" size="6"/>
 				</textElement>
-				<textFieldExpression><![CDATA["www.ne.ch/sgrf"]]></textFieldExpression>
-				<hyperlinkReferenceExpression><![CDATA["www.ne.ch/sgrf"]]></hyperlinkReferenceExpression>
+				<textFieldExpression><![CDATA[$R{furtherInfromationText}]]></textFieldExpression>
```
Comment (voisardf) [inline print/print-apps/oereb/toc.jrxml]: and here

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 10]
```diff
@@ -68,6 +68,14 @@ def __call__(self, value, system):
 
         self.convert_to_printable_extract(extract_as_dict, feature_geometry, pdf_to_join)
 
+        extract_as_dict['furtherInfromationText'] = Config.get(
```
Comment (voisardf) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: follow up :)
find & replace

File: pyramid_oereb/standard/pyramid_oereb.yml.mako [hunk 11]
```diff
@@ -84,6 +84,11 @@ pyramid_oereb:
     # The headers sent to the print
     headers:
       Content-Type: application/json; charset=UTF-8
+    furtherInfromationText:
```
Comment (voisardf) [inline pyramid_oereb/standard/pyramid_oereb.yml.mako]: last one?

File: pyramid_oereb/standard/pyramid_oereb.yml.mako [hunk 12]
```diff
@@ -84,6 +84,11 @@ pyramid_oereb:
...
+      de: [Hier sollten die weiteren Informationen erfasst werden.]
+      fr: [...]
```
Comment (voisardf) [inline pyramid_oereb/standard/pyramid_oereb.yml.mako]: fr: [Veuillez saisir les informations complémentaires ici]

**Comment by voisardf** [comment]
Seems I missed something there? Don't you wan't it to be handled in the ressources files?

**Comment by vvmruder** [comment]
I changed it to the configuration since @sbrunner suggestion is right. We do not want to change the print app only for a piece of text which varies between cantons.

**Comment by vvmruder** [comment]
:) @voisardf 
Thanks for reading all my confusing code :smile:

**Pull request merged**